lib/pack.js

```diff
@@ -1,363 +1,57 @@
-    .then(() => cacache.tmp.withTmp(npm.tmp, {tmpPrefix: 'unpacking'}, (tmp) => {
-      const tmpTarget = path.join(tmp, filename)
-      return pacote.extract(arg, tmpTarget, opts)
-        .then(() => readJson(path.join(tmpTarget, 'package.json')))
-    }))
-    .then((pkg) => getContents(pkg, target, filename))
-}
-
-module.exports.prepareDirectory = prepareDirectory
-function prepareDirectory (dir) {
-  return readJson(path.join(dir, 'package.json')).then((pkg) => {
-    if (!pkg.name) {
-      throw new Error('package.json requires a "name" field')
-    }
-    if (!pkg.version) {
-      throw new Error('package.json requires a valid "version" field')
-    }
-    if (!pathIsInside(dir, npm.tmp)) {
-      if (pkg.scripts && pkg.scripts.prepublish) {
-        prepublishWarning([
-          'As of npm@5, `prepublish` scripts are deprecated.',
-          'Use `prepare` for build steps and `prepublishOnly` for upload-only.',
-          'See the deprecation note in `npm help scripts` for more information.'
-        ])
-      }
-      if (npm.config.get('ignore-prepublish')) {
-        return lifecycle(pkg, 'prepare', dir).then(() => pkg)
-      } else {
-        return lifecycle(pkg, 'prepublish', dir).then(() => {
-          return lifecycle(pkg, 'prepare', dir)
-        }).then(() => pkg)
-      }
-    }
-    return pkg
-  })
-}
-
-module.exports.packDirectory = packDirectory
-function packDirectory (mani, dir, target, filename, logIt, dryRun) {
-  deprCheck(mani)
-  return readJson(path.join(dir, 'package.json')).then((pkg) => {
-    return lifecycle(pkg, 'prepack', dir)
-  }).then(() => {
-    return readJson(path.join(dir, 'package.json'))
-  }).then((pkg) => {
-    return cacache.tmp.withTmp(npm.tmp, {tmpPrefix: 'packing'}, (tmp) => {
-      const tmpTarget = path.join(tmp, path.basename(target))
-
-      const tarOpt = {
-        file: tmpTarget,
-        cwd: dir,
-        prefix: 'package/',
-        portable: true,
-        // Provide a specific date in the 1980s for the benefit of zip,
-        // which is confounded by files dated at the Unix epoch 0.
-        mtime: new Date('1985-10-26T08:15:00.000Z'),
-        gzip: true
-      }
-
-      return BB.resolve(packlist({ path: dir }))
-      // NOTE: node-tar does some Magic Stuff depending on prefixes for files
-      //       specifically with @ signs, so we just neutralize that one
-      //       and any such future "features" by prepending `./`
-        .then((files) => tar.create(tarOpt, files.map((f) => `./${f}`)))
-        .then(() => getContents(pkg, tmpTarget, filename, logIt))
-        // thread the content info through
-        .tap(() => {
-          if (dryRun) {
-            log.verbose('pack', '--dry-run mode enabled. Skipping write.')
-          } else {
-            return move(tmpTarget, target, {Promise: BB, fs})
-          }
-        })
-        .tap(() => lifecycle(pkg, 'postpack', dir))
+  const tarballs = await Promise.all(args.map((arg) => pack_(arg)))
```

Comment:
Since there's an `await` here, any errors will throw.  Probably want to wrap this whole thing in a `try { ... } catch (er) { cb(er) }`